python/servo/testing_commands.py

```diff
@@ -633,35 +671,67 @@ def speedometer_runner(self, binary: str, bmf_output: str | None):
         print(f"Score: {speedometer['Score']['mean']} ± {speedometer['Score']['delta']}")
 
         if bmf_output:
-            output = dict()
-
-            def parse_speedometer_result(result):
-                if result["unit"] == "ms":
-                    output[f"Speedometer/{result['name']}"] = {
-                        "latency": {  # speedometer has ms we need to convert to ns
-                            "value": float(result["mean"]) * 1000.0,
-                            "lower_value": float(result["min"]) * 1000.0,
-                            "upper_value": float(result["max"]) * 1000.0,
-                        }
-                    }
-                elif result["unit"] == "score":
-                    output[f"Speedometer/{result['name']}"] = {
-                        "score": {
-                            "value": float(result["mean"]),
-                            "lower_value": float(result["min"]),
-                            "upper_value": float(result["max"]),
-                        }
-                    }
-                else:
-                    raise "Unknown unit!"
+            self.speedometer_to_bmf(speedometer, bmf_output)
 
-                for child in result["children"]:
-                    parse_speedometer_result(child)
+    def speedometer_runner_ohos(self, bmf_output: str | None):
+        hdc_path: str = shutil.which("hdc")
+        log_path: str = "/data/app/el2/100/base/org.servo.servo/cache/servo.log"
+        if hdc_path is None:
+            hdc_path = path.join(os.getenv("OHOS_SDK_NATIVE"), "../", "toolchains", "hdc")
+
+        def read_log_file() -> str:
+            subprocess.call([hdc_path, "file", "recv", log_path])
+            file = ""
+            try:
+                file = open("servo.log")
+            except OSError:
+                return ""
+            return file.read()
+
+        subprocess.call([hdc_path, "shell", "aa", "force-stop", "org.servo.servo"])
+
+        subprocess.call([hdc_path, "shell", "rm", log_path])
+        subprocess.call(
+            [
+                hdc_path,
+                "shell",
+                "aa",
+                "start",
+                "-a",
+                "EntryAbility",
+                "-b",
+                "org.servo.servo",
+                "-U",
+                "https://servospeedometer.netlify.app?headless=1",
+                "--ps",
+                "--log-filter",
+                "script::dom::console",
+                "--psn",
+                "--log-to-file",
+            ]
+        )
 
-            for v in speedometer.values():
-                parse_speedometer_result(v)
-            with open(bmf_output, "w", encoding="utf-8") as f:
-                json.dump(output, f, indent=4)
+        # A current (2025-07-23) run took 3m 49s = 229s. We keep a safety margin
+        # but we will exit earlier if we see "{"
+        # Currently ohos has a bug where the event loop gets stuck. We produce a
```

Comment:
Can you create an issue for this? Are you sure it doesn't happen on desktop? E.g. if the mouse is outside the window area, so no interaction.

Reply:
https://github.com/servo/servo/issues/37727